@types/index.d.ts

```diff
@@ -131,52 +139,37 @@ interface Request extends Body {
 	 * Returns the URL of request as a string.
 	 */
 	readonly url: string;
-	clone: () => Request;
+	clone(): Request;
 }
-type RequestInfo = string | Body;
-declare var Request: {
-	prototype: Request;
-	new (input: RequestInfo, init?: RequestInit): Request;
-};
 
-interface Response extends Body {
+export class Response extends Body {
+	constructor(body?: BodyInit | null, init?: ResponseInit);
+
 	readonly headers: Headers;
 	readonly ok: boolean;
 	readonly redirected: boolean;
 	readonly status: number;
 	readonly statusText: string;
 	readonly url: string;
-	clone: () => Response;
-}
-
-declare var Response: {
-	prototype: Response;
-	new (body?: BodyInit | null, init?: ResponseInit): Response;
-};
-
-declare function fetch(url: RequestInfo, init?: RequestInit): Promise<Response>;
-
-declare namespace fetch {
-	function isRedirect(code: number): boolean;
+	clone(): Response;
 }
 
-interface FetchError extends Error {
+export class FetchError extends Error {
+	constructor(message: string, type: string, systemError?: object);
+	
 	name: 'FetchError';
 	[Symbol.toStringTag]: 'FetchError';
 	type: string;
 	code?: string;
 	errno?: string;
 }
-declare var FetchError: {
-	prototype: FetchError;
-	new (message: string, type: string, systemError?: object): FetchError;
-};
 
 export class AbortError extends Error {
 	type: string;
 	name: 'AbortError';
 	[Symbol.toStringTag]: 'AbortError';
 }
 
-export {Headers, Request, Response, FetchError};
-export default fetch;
+export function isRedirect(code: number): boolean;
+
+export default function fetch(url: RequestInfo, init?: RequestInit): Promise<Response>;
```

Comment:
Since the `main` field will point to `index.cjs`, and the `types` field (which is the corresponding field for the typings) will point to this file, I think that the correct thing here would be to use a CommonJS export:

```suggestion
declare function fetch(url: RequestInfo, init?: RequestInit): Promise<Response>;
export = fetch;
```

You can read some more info it here: https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default

Reply:
If you look at the built `index.cjs`, the resulting `exports` are the following:
```javascript
exports.default = fetch;
```

So the good way to type it is to use `export default`.
`export = fetch` is equivalent to `module.exports = fetch` which is not the case here...

Reply:
ahh, cool, I was looking at the currently published package which uses `module.exports = fetch`, my bad!

Reply:
@SomaticIT I just cloned the repo and made a clean build, and the output in the bundle is `module.exports = fetch`, thus I think that we should type this as `export = fetch`